src/main/java/com/gregtechceu/gtceu/api/data/chemical/material/ItemMaterialData.java

```diff
@@ -173,29 +173,43 @@ public static void reinitializeMaterialData() {
     public static void resolveItemMaterialInfos(Consumer<FinishedRecipe> provider) {
-        for (var entry : UNRESOLVED_ITEM_MATERIAL_INFO.entrySet()) {
-            List<MaterialStack> stacks = new ArrayList<>();
+        for (var iter = UNRESOLVED_ITEM_MATERIAL_INFO.entrySet().iterator(); iter.hasNext();) {
+            var entry = iter.next();
             var stack = entry.getKey();
-            int outputCount = stack.getCount();
-            for (var input : entry.getValue()) {
-                var matStack = getMaterialInfo(input.getItem());
-                int inputCount = input.getCount();
-                if (matStack != null) {
-                    matStack.getMaterials()
-                            .forEach(ms -> stacks.add(ms.multiply(inputCount).divide(outputCount)));
-                }
+            var existingMaterialInfo = ITEM_MATERIAL_INFO.get(stack.getItem());
+            recurseFindMaterialInfo(existingMaterialInfo, stack);
+            if (existingMaterialInfo != null) {
+                RecyclingRecipes.registerRecyclingRecipes(provider, stack.copyWithCount(1),
+                        existingMaterialInfo.getMaterials(), false, null);
             }
-            if (stacks.isEmpty()) continue;
-            var matInfo = ITEM_MATERIAL_INFO.get(stack.getItem());
-            if (matInfo == null) {
-                matInfo = new ItemMaterialInfo(stacks);
-                ITEM_MATERIAL_INFO.put(stack.getItem(), matInfo);
+            iter.remove();
+        }
+    }
+
+    private static void recurseFindMaterialInfo(ItemMaterialInfo info, ItemStack stack) {
+        // grab material info from each input
+        for (var input : UNRESOLVED_ITEM_MATERIAL_INFO.get(stack)) {
+            // recurse if its nested inputs, not yet resolved
+            if (UNRESOLVED_ITEM_MATERIAL_INFO.containsKey(input)) {
+                recurseFindMaterialInfo(info, input);
             } else {
-                matInfo.addMaterialStacks(stacks);
+                // add the info from an item that is resolved (or not in the map to begin with)
+                var singularMatInfo = getMaterialInfo(input.getItem());
+                int inputCount = input.getCount();
+                int outputCount = stack.getCount();
+                if (singularMatInfo != null) { // if that material info exists
+                    List<MaterialStack> stackList = new ArrayList<>();
+                    for (var matStack : singularMatInfo.getMaterials()) {
+                        stackList.add(matStack.multiply(inputCount).divide(outputCount));
+                    }
+                    if (info == null) { // if the info isn't set initialize it
+                        info = new ItemMaterialInfo(stackList);
+                        ITEM_MATERIAL_INFO.put(stack.getItem(), info);
+                    } else { // otherwise, add to it
+                        info.addMaterialStacks(stackList);
+                    }
+                }
```

Comment:
Maybe add an else block here that logs a warning?
since an item not being in UNRESOLVED_ITEM_MATERIAL_INFO and also not having a getMaterialInfo(stack) now gets quietly ignored, but that should never happen I think

Reply:
no warning is needed, it still needs to add the item to the recipe, it just wont contribute anything to decomp